src/routes/tools/_common/helpers.js

```diff
@@ -308,22 +303,39 @@ export const getInitialConfig = (
 };
 
 /**
- * Create initial location
- * @param {float} lng
- * @param {float} lat
- * @param {string} boundary - boundary id
- * @return {object}
+ * setInitialLocation: Queries the Cal-Adapt API for boundary type feature(s).
+ * Typically used to create an object with initial location for a Cal-Adapt tool.
+ * @param {float} lng - feature's centroid longitude coordinate
+ * @param {float} lat - feature's centroid latitude coordinate
+ * @param {string} boundary - boundary type, e.g. "locagrid", "counties", etc.
+ * @param {number} featureId - unique id of location feature
+ * @return {object} - GeoJSON feature or feature collection on success
  */
-// Helper function to create an object with initial location for a Cal-Adapt tool
-export async function setInitialLocation(lng, lat, boundary) {
+export async function setInitialLocation(lng, lat, boundary, featureId) {
   let loc = DEFAULT_LOCATION;
 
-  try {
-    loc = await getFeature({ center: [lng, lat] }, boundary);
-  } catch (error) {
-    console.warn(error);
+  if (isValidNumber(featureId)) {
+    try {
+      loc = await getFeatureById(boundary, featureId);
+    } catch (error) {
+      console.warn(error);
+    }
+  } else if (isValidNumber(lng) && isValidNumber(lat)) {
+    // Prior to PR#235 feature data was retrieved this way, but it is error prone.
+    // For more info, see: https://trello.com/c/8JmopK9Q
+    // NOTE: this code still exists in case a legacy bookmarked URL that only
+    // has the lng,lat coords and not the featureId.
+    try {
+      loc = await getFeature({ center: [lng, lat] }, boundary);
+    } catch (error) {
+      console.warn(error);
+    }
+  } else {
+    return loc;
   }
 
+  // FIXME: locagrid should use its own title which consists of the lng, lat of
```

Comment:
See about adding the `locagrid` title here, now that it has been added to the API.

Reply:
Maybe address this in a separate PR as this one has already grown to be a little big.